### ISSUE TITLE
molecular library

### ISSUE THREAD

**Issue by qzhu2017** [qzhu2017/PyXtal#18]
Instead of using the molecular libraries in ASE, we need to build a library for our own studies. I will start to collect such examples later.

**Comment by qzhu2017** [comment]
Three ways to allow the users to input the molecular information

1, standard xyz file
2, through an internal database (by our own library with human readable name, like h20, ch4, etc), which is similar to ase.
3, to obtain the geometry xyz file from on-line database like [pubchem](https://pubchem.ncbi.nlm.nih.gov).

One more option:
Automatically detect the flexibility and allow the variation of flexible dihedral angles (maybe we can do it with Bible?)

**Comment by qzhu2017** [comment]
Finish 2, 3
77a0f98

**Comment by scottfredericks** [comment]
@qzhu2017 
For the issue of flexibility, is the z_matrix file format an option?

**Comment by qzhu2017** [comment]
Yes, z-matrix is a possible option. I believe there exist some other well developed libraries which can handle such charity and flexibility issues. Something like openbabel or something else.

**Comment by scottfredericks** [comment]
@qzhu2017 
I have updated occurrences of ASE to openbabel, and have made the corresponding changes in the documentation and setup.py.

By default, I have openbabel set to read strings using the SMILES format:
https://en.wikipedia.org/wiki/Simplified_molecular-input_line-entry_system
However, it looks like you already have a system for getting molecules from a database. It seems like this method would be better for when we input molecules using strings like "H2O" or "CH4". I will read through your code and implement your function in these places instead of openbabel.

**Comment by scottfredericks** [comment]
@qzhu2017 
I changed molecular_crystal and molecular_crystal_2D, so that they now try to access database.collection molecules first. If no matching string is found, they will then interpret the string using the SMILES format. I think there might be a better format, which is easier for the user. But I have not gotten a chance to look through all of the openbabel formats yet, since there are so many of them. If you know of another string-like format that would be better, just let me know and I can change the default format. Otherwise, I will put this issue on hold and focus on troubleshooting the 2d crystals.

**Comment by scottfredericks** [comment]
@qzhu2017 
It seems that regardless of what input string format we use, we will still need to optimize the generated molecule. By default, openbabel just sets all coordinates to (0,0,0). There seem to be force-field functions for finding the correct points, but I am still looking for the correct implementation. The documentation uses a confusing mixture of both the openbabel Python bindings, and also the pybel library, which is slightly different. So, I have been unable to figure out how to get the coordinates without using Pybel (which I cannot figure out how to install, since I already have a different library called Pybel installed)

**Comment by qzhu2017** [comment]
@scottfredericks , as we discussed, don't forget to make openbabel as an optional package.
It is sometimes not easy to install this package.

I will add some documentation regard the installation of openbabel/swig. 

For SMILE->xyz in openbabel
https://openbabel.org/docs/dev/Command-line_tools/babel.html
it seems that an option of --gen3d is needed.

If you want to use pubchem, you can just look at my script.
https://github.com/qzhu2017/PyXtal/blob/master/pyxtal/test_cases/get_molecule_from_pubchem.py
below is an example of searching by name, I think you can easily change it to SMILES format
```
mol = pcp.get_compounds(name, 'name', record_type='3d')[0]
```

**Comment by scottfredericks** [comment]
@qzhu2017 
I have made openbabel optional. When loading molecules within molecular_crystal, we try, in the following order:
1. Load from collections.py
2. Load from a file (using mol_from_file, which is in molecule.py)
3. Load from a string (using mol_from_string, which is also in molecule.py)
Openbabel still adds extra file formats for option (3), but most users will probably not need it. I have added these changes to the documentation.
